parlai/crowdsourcing/utils/tests.py

```diff
@@ -308,22 +349,18 @@ def _test_agent_states(
                     'episode_done': False,
                 },
             )
+            self.await_channel_requests()
 
         # Submit the HIT
         for agent_id in agent_ids:
-            self.server.send_agent_act(
-                agent_id=agent_id,
-                act_content={
-                    'task_data': {'final_data': {}},
-                    'MEPHISTO_is_submit': True,
-                },
-            )
+            self.server.submit_mock_unit(agent_id, {'final_data': {}})
+            self.await_channel_requests()
 
         # # Check that the inputs and outputs are as expected
 
         # Wait until all messages have arrived
         wait_time = 5.0  # In seconds
-        max_num_tries = 30  # max_num_tries * wait_time is the max time to wait
+        max_num_tries = 1  # max_num_tries * wait_time is the max time to wait
```

Comment:
Hmm - so we should just be able to get rid of `max_num_tries` and the retry code altogether, right?

Reply:
Almost definitely yes, but I was focused on minimal changes.